### ISSUE TITLE
libev connection: RuntimeError: dictionary changed size during iteration

### ISSUE THREAD

**Issue by ztane** [pika/pika#595]
This happens about every time:

```
    for timer in self._active_timers:
RuntimeError: dictionary changed size during iteration
```

This is because [libev_connection.py:161](https://github.com/pika/pika/blob/master/pika/adapters/libev_connection.py#L161) does not copy the keys as list: (this possibly should read `for timer in list(self._active_timers):`).

**Comment by vitaly-krugl** [comment]
+1

**Comment by ztane** [comment]
it is surprising that the test suite actually does *not* test these.

**Comment by vitaly-krugl** [comment]
The project could really use more contributions from the community to improve the test coverage situation.

I added a bunch of BlockingConnection acceptance tests over the weekend, but much more is needed in the various adapters and core code, including extensive unit tests.

**Comment by vitaly-krugl** [comment]
@ztane, this wsa fixed by PR #672

**Comment by vitaly-krugl** [comment]
fixed by PR #672
